### PR TITLE
Auto-assume service account in installation repo

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
If the owner's email of the project is not the current user, impersonate that user and update go client.

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: cmd/plural/validation.go [hunk 1]
```diff
@@ -192,3 +161,36 @@ func upstreamSynced(fn func(*cli.Context) error) func(*cli.Context) error {
...
 	}
 }
+
+func (p *Plural) impersonate(fn func(*cli.Context) error) func(*cli.Context) error {
+	return func(c *cli.Context) error {
+		p.InitPluralClient()
+		path := manifest.ProjectManifestPath()
+		project, err := manifest.ReadProject(path)
+		conf := config.Read()
+		if err != nil {
```
Comment (michaeljguarino) [inline cmd/plural/validation.go:171]: We probably shouldn't error here, or is it only used when the repo is already set up?

**Review by michaeljguarino — COMMENTED** [review]
One thing we need to add here is there are other fetches to the config w/in the code outside client initialization.  The biggest I can think of is scaffolding terraform and helm but we should double check other possibilities too

**Comment by michaeljguarino** [comment]
I think if when we assume a service account, a global variable within the `pkg/config` package is set and any future `config.Read()` function returns it if non-null it might guarantee all code paths are auto-wired with the service account.  Or at least that's how i guessed an elegant solve might work

File: cmd/plural/validation.go (hunk 1 above)
Comment (zreigz) [reply cmd/plural/validation.go]: yes, it's used when the repo is already set up

**Review by michaeljguarino — APPROVED** [review]
lgtm

**Pull request merged**